include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -589,11 +595,14 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
                             break;
 
                         default:
-                            throw format_error{std::string{"Illegal SAM header tag starting with: "} + *it};
+                            seqan3::debug_stream << "Not supported SAM header tag in @PG starting with: " << *it << '\n';
                     }
 
                     parse_tag_value(*who);
                 }
+                if (!id_tag_given)
```

Comment:
```suggestion
                if (tmp.id.empty())
```

Then you do not need the `id_tag_given`